### DISCUSSION TITLE
"join meeting" twice for unauthenticated users

### DISCUSSION THREAD

**Discussion by uhillebrand** [nordeck/jitsi-keycloak-adapter#23] [Q&A] [answered]
Hi! We are using the Adapter sucessfully for 2 of our Clients. Everything seems to work flawlessly, thanks a lot.

However, there is a minor issue: If I try to create a new meeting while not yet being authenticated, I hit the prejoin page at first (where I can set up mic, camera etc.). After selecting "join meeting", I am redirected to Keycloak to sign in. After authenticating, I get redirected back to the prejoin page again, where I have to select "join meeting" for the 2nd time.

Is there a way to get rid of having to click "join meeting" twice?

Thanks in advance!

**Comment by emrahcom** [comment, marked as answer]
Hi @uhillebrand,

I think it is possible by customizing `/usr/share/jitsi-meet/body.html`. The Javascript codes in this file catches the login button and redirects the user to Keycloak. You may embed an additional parameter to the return URL (_which is `config.prejoinConfig.enabled`_) to bypass the prejoin page after authentication.

So, find the following line in `/usr/share/jitsi-meet/body.html`:
It is line `#10` in [the current template](https://github.com/nordeck/jitsi-keycloak-adapter/blob/main/templates/usr/share/jitsi-meet/body.html#L10C5-L10C72)

```javascript
    const hash = encodeURIComponent(window.location.hash.substring(1));
```

and change it like the following:

```javascript
    //const hash = encodeURIComponent(window.location.hash.substring(1));
    let hash = "config.prejoinConfig.enabled=false";
    if (window.location.hash) {
      hash = `${hash}&${window.location.hash.substring(1)}`;
    }
    hash =  encodeURIComponent(hash);
```

**Reply by uhillebrand** [reply]
Hi @emrahcom,

this works like a charm, thank you.

Maybe this is a candidate for changing this in body.html from this project, as we copy a modified body.html anyway during installation?

Anyway, we now know how to fix this by ourselves, thanks again!

**Reply by emrahcom** [reply]
> Maybe this is a candidate for changing this in body.html from this project, as we copy a modified body.html anyway during installation?

Agree, I'm planning to add it after some more testing.

**Reply by uhillebrand** [reply]
Perfect. If there are problems on our end we'll let you know.
